### PR TITLE
Feature/zipkin support

### PR DESCRIPTION
**Pull request by fibbery**
### Issues associated with this PR

Your PR should present related issues you want to solve.

### Solutions
You should show your solutions about the issues in your PR, including the overall solutions, details and the changes. At this time, Chinese is allowed to describe these.

### UT result
Unit Test is needed if the code is changed, your unit test should cover boundary cases, corner cases, and some exceptional cases. And you need to show the UT result.

### Benchmark
If your code involves the processing of every request, you should give the Benchmark Result.

### Code Style
+ Make sure `Goimports` has run
+ Show `Golint` result

### CODE REVIEW

File: pkg/trace/zipkin/driver.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	"mosn.io/mosn/pkg/trace"
+	"mosn.io/mosn/pkg/types"
+)
+
+const (
+	DriverName = "Zipkin"
+)
+
+func init() {
+	trace.RegisterDriver(DriverName, NewZipkinDriverImpl())
```
Comment (nejisama) [inline pkg/trace/zipkin/driver.go:33]: 不要使用init 进行注册了，改成在main里显式调用。main下有一个ExtensionsRegister 函数做类似的事情

File: pkg/config/v2/zipkin.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+ * limitations under the License.
+ */
+
+package v2
+
+import (
+	"errors"
+)
+
+const (
```
Comment (nejisama) [inline pkg/config/v2/zipkin.go:24]: 相关的配置也可以保留到trace/zipkin 目录下，作为纯粹的扩展实现，不放到config下

File: cmd/mosn/main/mosn.go [hunk 3]
```diff
@@ -69,6 +69,7 @@ import (
 	_ "mosn.io/mosn/pkg/trace/sofa/http"
 	_ "mosn.io/mosn/pkg/trace/sofa/xprotocol"
 	_ "mosn.io/mosn/pkg/trace/sofa/xprotocol/bolt"
+	_ "mosn.io/mosn/pkg/trace/zipkin"
```
Comment (nejisama) [inline cmd/mosn/main/mosn.go:72]: 通过显式注册的方式，代替init的方式；同时修改main需要同时修改istio目录下两个main

File: pkg/trace/zipkin/tracer.go [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+	"mosn.io/mosn/pkg/protocol"
+	"mosn.io/mosn/pkg/trace"
+	"mosn.io/mosn/pkg/types"
+
+	"github.com/openzipkin/zipkin-go"
+	"github.com/openzipkin/zipkin-go/model"
+)
+
+func init() {
+	trace.RegisterTracerBuilder(DriverName, protocol.HTTP1, NewHttpTracer)
```
Comment (doujiang24) [inline pkg/trace/zipkin/tracer.go:45]: we do not need this line now? since we already register it in `control.go`?

**Review by doujiang24 — COMMENTED** [review]
Already need to register it in 
https://github.com/mosn/mosn/blob/c051b22b6ddfe1c22f089dc5165c43aae0da6040/istio/_istio152/main/control.go
otherwise, LGTM.

**Review by doujiang24 — COMMENTED** [review]
LGTM.

**Review by doujiang24 — APPROVED** [review]

**Review by taoyuanyuan — APPROVED** [review]
LGTM, thank you for your contribution~

**Pull request merged**